### DISCUSSION TITLE
Create cable (and cablepath) from script

### DISCUSSION THREAD

**Discussion by dainok** [netbox-community/netbox#10199] [Q&A]
Hi! I'm working on netbox v3.3.2 . I need to create cables from a discovery script. I can create cable from device, interfaces, as following:

~~~
c = Cable.objects.create()
int_a = Interface.objects.get(device=1, id=3)
int_b = Interface.objects.get(device=6, id=32)
cbt1 = CableTermination.objects.create(termination = int_a, cable=c, cable_end="A")
cbt2 = CableTermination.objects.create(termination = int_b, cable=c, cable_end="B")
~~~

I cannot find the way to create the CablePath. I looked at doc pages, sources, forums... but I still have no ideas. I need a simple cablepath int_a -> int_b (and back).

Anyone can point me to the right direction?
Thank you

**Comment by jeremystretch** [comment]
CablePaths are created automatically by NetBox as needed; you don't need to create them directly.

**Comment by dainok** [comment]
If I force the save() the CablePath is still empty. I guess I'm missing a step.

~~~
Cable.objects.all() # -> empty
CablePath.objects.all() # -> empty

c = Cable.objects.create()
int_a = Interface.objects.get(device=1, id=3)
int_b = Interface.objects.get(device=6, id=32)
cbt1 = CableTermination.objects.create(termination = int_a, cable=c, cable_end="A")
cbt2 = CableTermination.objects.create(termination = int_b, cable=c, cable_end="B")
c.save()

Cable.objects.all() # -> contains 1 cable
CablePath.objects.all() # -> still empty
~~~

**Reply by kkthxbye-code** [reply]
```python
int_a = Interface.objects.get(device=1, id=3)
int_b = Interface.objects.get(device=6, id=32)
c = Cable(a_terminations=[int_a], b_terminations=[int_b])
c.save()
```

If you wanna keep doing it the way you were, you would need to set `c._terminations_modified = True` and then save the Cable I think.

**Reply by dainok** [reply]
Perfect, thank you! Exactly what I was looking for.

**Reply by candlerb** [reply]
As an aside: you need to write

```
c.full_clean()
c.save()
```

If you don't do this, the Django ORM will let you save all sorts of junk that doesn't pass model validation.  I've been bitten by this several times before, and IMO it's a big [mis-feature](https://stackoverflow.com/questions/4441539/why-doesnt-djangos-model-save-call-full-clean) of the ORM.

**Reply by dainok** [reply]
Added, thank you. I'm close to release [netdoc](https://github.com/dainok/netdoc)

**Reply by candlerb** [reply]
> Added, thank you. I'm close to release [netdoc](https://github.com/dainok/netdoc)

Suggestion: add a brief description of what your plugin actually *does*.  So far, all I can see is "An automatic network documentation plugin for NetBox"
